pycbc/inference/models/relbin.py

```diff
@@ -881,3 +898,13 @@ def _loglr(self):
         else:
             results = loglr
         return results
+
+    def _nowaveform_handler(self):
+        """Sets loglr values if no waveform generated.
+        """
+        loglr = sh_total = hh_total = -numpy.inf
+        if self.return_sh_hh:
+            results = (sh_total, hh_total)
```

Comment:
I'm not sure it makes sense to catch the case where the return products are different. These should never be called in such a case and you can't be certain you are matching the return product format (e.g. is it a vector or a scalar?). This path should never be used for normal calling of the model (e.g. only reconstruction or interaction with other model which should itself probably decide what to do with its own handler rather than work out how to handle unexpected input.

Reply:
I was just trying to match the return signature. What do you suggest doing in this case? Just return -numpy.inf?

Reply:
Yeah, I think that case should just be an uncaught error e.g. it probably *should* blow up on the user since I think they are doing something wrong to end up in that situation.